src/nyc_trees/nyc_trees/middleware.py

```diff
@@ -25,3 +26,13 @@ def process_view(self, request, view_func, view_args, view_kwargs):
 
         if not allowed:
             return redirect(self.redirect_url)
+
+
+# Author: Tony Abou-Assaleh
+# Source: http://stackoverflow.com/a/7871831
```

Comment:
I was just going to redirect to the logout page at first, but got the idea to return the `contrib.auth` logout view from this person on SO.